libs/components/modals/src/lib/modules/modal/modal.component.spec.ts

```diff
@@ -1133,4 +1135,118 @@ describe('Modal component', () => {
       document.querySelector('.sky-modal-dialog')
     ).toBeAccessible();
   });
+
+  describe('Is dirty directive', () => {
+    function getConfirmModalElement(): HTMLElement | null {
+      return document.querySelector('.sky-confirm');
+    }
+
+    function getDiscardButtonElement(
+      confirmModalEl: HTMLElement
+    ): HTMLButtonElement | null {
+      return confirmModalEl.querySelector('.sky-btn-primary');
+    }
+
+    function getKeepWorkingButtonElement(
+      confirmModalEl: HTMLElement
+    ): HTMLButtonElement | null {
+      return confirmModalEl.querySelector('.sky-btn-link');
+    }
+
+    async function checkConfirmModalIsCorrect(
+      confirmModalEl: HTMLElement
+    ): Promise<void> {
+      const messageEl = confirmModalEl.querySelector('.sky-confirm-message');
+      await expectAsync(messageEl).toHaveLibResourceText(
+        'skyux_modal_dirty_default_message'
+      );
+
+      const discardButtonEl = getDiscardButtonElement(confirmModalEl);
+      await expectAsync(discardButtonEl).toHaveLibResourceText(
+        'skyux_modal_dirty_default_discard_changes_text'
+      );
+
+      const keepWorkingButtonEl = getKeepWorkingButtonElement(confirmModalEl);
+      await expectAsync(keepWorkingButtonEl).toHaveLibResourceText(
+        'skyux_modal_dirty_default_keep_working_text'
+      );
+    }
+
+    let dirtyContextProvider: { providers: StaticProvider[] };
+
+    beforeEach(fakeAsync(() => {
```

Comment:
You don't need `fakeAsync()` here. Also verify that it's needed in other unit tests as using it unnecessarily makes it harder to refactor/modify tests.

Reply:
Good catch. That was left behind from some refactoring I was doing. As for the other tests, you might be right in it not being needed. I thought it was necessary because of the `closeModal` function, but that only calls the function `tick()` if you pass in `true` for `async`. Otherwise, it ticks the ApplicationRef.